pr_check.sh

```diff
@@ -1,10 +1,37 @@
 #!/bin/bash
 
+# --------------------------------------------
+# Options that must be configured by app owner
+# --------------------------------------------
+APP_NAME="xjoin"  # name of app-sre "application" folder this component lives in
+COMPONENT_NAME="xjoin-operator"  # name of app-sre "resourceTemplate" in deploy.yaml for this component
+IMAGE="quay.io/cloudservices/xjoin-operator"  # image location on quay
+
+IQE_PLUGINS=pr_check"xjoin"  # name of the IQE plugin for this app.
+IQE_MARKER_EXPRESSION="smoke"  # This is the value passed to pytest -m
+IQE_FILTER_EXPRESSION=""  # This is the value passed to pytest -k
+IQE_CJI_TIMEOUT="30m"  # This is the time to wait for smoke test to complete or fail
+
+DOCKERFILE="Dockerfile.unittest"
```

Comment:
is a separate Dockerfile needed? Could you do something like this instead? https://github.com/RedHatInsights/xjoin-validation/blob/master/unit_test.sh#L18

Reply:
Thanks to using `distroless` image use, there are no utilities available to execute anything.   Here is what I just tried:
```
[aarif@daamaan ~]$ podman run --entrypoint=""  quay.io/cloudservices/xjoin-operator:bd4d02d ginkgo ./...
Error: crun: executable file `ginkgo` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found
[aarif@daamaan ~]$ 
[aarif@daamaan ~]$ 
[aarif@daamaan ~]$ 
[aarif@daamaan ~]$ podman run --entrypoint=""  quay.io/cloudservices/xjoin-operator:bd4d02d ls -al
Error: crun: executable file `ls` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found
[aarif@daamaan ~]$ 
[aarif@daamaan ~]$ 
[aarif@daamaan ~]$ podman run --entrypoint=""  quay.io/cloudservices/xjoin-operator:bd4d02d bash -c "ls -al"
Error: crun: executable file `bash` not found in $PATH: No such file or directory: OCI runtime attempted to invoke a command that was not found
[aarif@daamaan ~]$ 
```
I can investigate how to run tests using the `xjoin-operator` image.  That said, using `Dockerfile.unittest` is quick and simple unless it poses security risk.

Reply:
ok - if it's too much trouble the tests can be run outside of docker. Currently it looks like in the unit_test.sh file the tests will run directly on Jenkins which is fine for now.

```
# use the following make rule to execute unit tests
make generic-test
result=$?
set -e
```

Reply:
Yes it appears to be.